common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/Mappings.scala

```diff
@@ -188,7 +188,9 @@ object Mappings {
     metadataMapping("metadata"),
     usageRightsMapping("usageRights"),
     photoshootMapping("photoshoot"),
-    dateField("lastModified")
+    dateField("lastModified"),
+    dateField("deletedAt"),
+    keywordField("deletedBy")
```

Comment:
I don't think this is the correct location for these fields as it suggests this is the time the user metadata was deleted, rather than the image itself, which is what we want to represent. A better location would be at the [root of the document as siblings to `uploadTime` and `uploadedBy`](https://github.com/guardian/grid/blob/c7ccb0ab6603209aaa816188e2547fb7394d90af/common-lib/src/main/scala/com/gu/mediaservice/lib/elasticsearch/Mappings.scala#L53-L54).

We may also want to call it `deletedTime` for consistency?

Reply:
@akash1810 yes it will be making more sense 
will change the name and move it to the `Image` model

Reply:
@akash1810 I couldn't add `deleteTime` and `deletedBy` at the root of the document  because we hit  function parameters maximum number `implementation restricts functions to 22 parameters`
so I added them into a nested object `softDeletedMetadata`

Reply:
Ah, Scala, such a wonderful language 😉 .